src/assets/scripts/socket.js

```diff
@@ -0,0 +1,60 @@
+const maxRetries = 30;
+
+export default class Socket {
+  constructor() {
+    this.retries = 0;
+    this._receivers = [];
+  }
+
+  _establishConnection() {
+    return new Promise((resolve, reject) => {
+      const connection = new WebSocket(`ws://${location.host}`);
+      connection.on('open', () => {
+        this.retries++;
+        this._listen();
+        resolve();
+      });
+      connection.on('error', error => {
+        error = error || new Error('Connect could not be established.');
+        if (this.retries < maxRetries) {
```

Comment:
Didn't think of this before, but this logic should be similar for the peer connection as well, maybe we should pull it out and use it for both.